.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,59 @@
+name: "Build and publish Docker images to GHCR"
+
+on:
+    push:
+        branches:
+            - "main"
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}-${{ github.job }}
+    cancel-in-progress: true
+
+permissions:
+    contents: read
+    packages: write
+
+jobs:
+    build-and-push:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                component: ["front", "api", "db"]
+        steps:
+            - name: PrepareReg Names
+              run: |
+                  echo IMAGE_REPOSITORY=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]') >> $GITHUB_ENV
+                  echo IMAGE_TAG=$(echo ${{ github.ref }} | tr '[:upper:]' '[:lower:]' | awk '{split($0,a,"/"); print a[3]}') >> $GITHUB_ENV
```

Comment:
I wonder if we can inject a command in `github.ref`. If we can, what are the implications ? Can we make a challenge out of this ? :-)

Reply:
If you change your name to some command injection payload, or someone forks CTFNote and changes the repository name to some command injection payload, I think it is definitely possible to attack yourself through your own repository (but not the other forks or upstream)! ;-)